indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -422,6 +425,45 @@ def _process(self):
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
 
 
+class RHBookingOccurrenceLink(RHBookingBase):
+    """Link a reservation occurrence to an event."""
+
+    @use_kwargs({
+        'date': fields.Date(required=True),
+    }, location='view_args')
+    @use_kwargs({
+        'admin_override_enabled': fields.Bool(load_default=False),
+    }, location='query')
+    def _process_args(self, date, admin_override_enabled):
+        RHBookingBase._process_args(self)
+        self.occurrence = self.booking.occurrences.filter_by(date=date).options(joinedload('link')).one()
+        self.event = Event.get_or_404(request.view_args['event_id'], is_deleted=False)
+        self.admin_override_enabled = admin_override_enabled
+
+    def _check_access(self):
+        RHBookingBase._check_access(self)
+        if self.occurrence.link is not None:
+            raise Forbidden('This booking occurrence is already linked')
+        if not self.occurrence.can_link(session.user, allow_admin=self.admin_override_enabled):
```

Comment:
This was not taking admin override mode into account, so an admin was always able to link any booking, instead of the expected behavior of only being able to do so when enabling Admin Override mode (ctrl-shift-a or via the menu). I only added the backend logic, what's still missing is:

- to implement the canLink logic properly on the frontend as well
- to send the `admin_override_enabled` flag with the linking request

I should be able to implement this fairly quickly tomorrow and merge the PR afterwards.

Reply:
This is now fixed as well and also cleaned up a few things I noticed on the way (e.g. not passing data through multiple method calls instead of just getting them from state/props where needed).